packages/editor-ui/src/utils/__tests__/nodeTypesUtils.spec.ts

```diff
@@ -0,0 +1,283 @@
+			},
+			type: 'n8n-nodes-base.googleSheets',
+			typeVersion: 4.5,
+			position: [440, 0],
+			id: '9a95ad27-06cf-4076-af6b-52846a109a8b',
+			name: 'Google Sheets',
+			credentials: {
+				googleSheetsOAuth2Api: {
+					id: '8QEpi028oHDLXntS',
+					name: 'milorad@n8n.io',
+				},
+			},
+		},
+		expected: ['Edit Fields', 'Edit Fields 2'],
+	},
+	{
+		caseName: 'Should only add one reference for each referenced node',
+		node: {
+			parameters: {
+				authentication: 'oAuth2',
+				resource: 'sheet',
+				operation: 'read',
+				documentId: {
+					__rl: true,
+					value: "={{ $('Edit Fields').item.json.document }}",
+					mode: 'id',
+				},
+				sheetName: {
+					__rl: true,
+					value: "={{ $('Edit Fields').item.json.sheet }}",
+					mode: 'id',
+				},
+				filtersUI: {},
+				combineFilters: 'AND',
+				options: {},
+			},
+			type: 'n8n-nodes-base.googleSheets',
+			typeVersion: 4.5,
+			position: [440, 0],
+			id: '9a95ad27-06cf-4076-af6b-52846a109a8b',
+			name: 'Google Sheets',
+			credentials: {
+				googleSheetsOAuth2Api: {
+					id: '8QEpi028oHDLXntS',
+					name: 'milorad@n8n.io',
+				},
+			},
+		},
+		expected: ['Edit Fields'],
+	},
+	{
+		caseName: 'Should handle multiple node references in one expression',
+		node: {
+			parameters: {
+				curlImport: '',
+				method: 'GET',
+				url: "={{ $('Edit Fields').item.json.one }} {{ $('Edit Fields 2').item.json.two }} {{ $('Edit Fields').item.json.three }}",
+				authentication: 'none',
+				provideSslCertificates: false,
+				sendQuery: false,
+				sendHeaders: false,
+				sendBody: false,
+				options: {},
+				infoMessage: '',
+			},
+			type: 'n8n-nodes-base.httpRequest',
+			typeVersion: 4.2,
+			position: [220, 220],
+			id: 'edc36001-aee7-4052-b66e-cf127f4b6ea5',
+			name: 'HTTP Request',
+		},
+		expected: ['Edit Fields', 'Edit Fields 2'],
+	},
+	{
+		caseName: 'Should ignore spaces around node references',
+		node: {
+			parameters: {
+				curlImport: '',
+				method: 'GET',
+				url: "={{ $('   Edit Fields     ').item.json.one }}",
+				authentication: 'none',
+				provideSslCertificates: false,
+				sendQuery: false,
+				sendHeaders: false,
+				sendBody: false,
+				options: {},
+				infoMessage: '',
+			},
+			type: 'n8n-nodes-base.httpRequest',
+			typeVersion: 4.2,
+			position: [220, 220],
+			id: 'edc36001-aee7-4052-b66e-cf127f4b6ea5',
+			name: 'HTTP Request',
+		},
+		expected: ['Edit Fields'],
+	},
+	{
+		caseName: 'Should ignore non-existing nodes',
+		node: {
+			parameters: {
+				curlImport: '',
+				method: 'GET',
+				url: "={{ $('   Edit Fields     ').item.json.one }}",
+				authentication: 'none',
+				provideSslCertificates: false,
+				sendQuery: false,
+				sendHeaders: false,
+				sendBody: false,
+				options: {},
+				infoMessage: '',
+			},
+			type: 'n8n-nodes-base.httpRequest',
+			typeVersion: 4.2,
+			position: [220, 220],
+			id: 'edc36001-aee7-4052-b66e-cf127f4b6ea5',
+			name: 'HTTP Request',
+		},
+		expected: ['Edit Fields'],
```

Comment:
this seems wrong. the node name here is `     Edit Fields     `, I would expect returned value to be `     Edit Fields     `. The space is part of the node name. If node name is `Edit Fields`, then this expression should not evaluate. 

this is different from `$(  'Edit Fields'  )` that I suggested before where the space is ignored because it's outside the name

Reply:
True, will update